src/common/wallet-state.ts

```diff
@@ -368,11 +368,13 @@ function useWalletState(initialState?: Partial<WalletStateParams>): WalletData {
     setTotalBalance(some(balance))
   }
 
-  const getTimestamp = async (blockNumber: number | null): Promise<Date | null> => {
-    if (blockNumber === null) return null
-    const {timestamp} = await web3.eth.getBlock(blockNumber, true)
-    return _.isString(timestamp) ? fromUnixTime(parseInt(timestamp, 16)) : fromUnixTime(timestamp)
-  }
+  const getTimestamp = _.memoize(
+    async (blockNumber: number | null): Promise<Date | null> => {
+      if (!blockNumber) return null
```

Comment:
This also includes a case when `blockNumber  === 0`, is that desired?

Reply:
:man_facepalming: